cmd/injector/main.go

```diff
@@ -17,95 +16,165 @@ import (
+	exitSuccess = 0
+)
+
+//nolint:gochecknoglobals // ok for providing as version output
+var (
+	// Version is the descriptive version, normally the tag from which the app was built.
+	// Since git tags can be changed, use Commit instead as the most accurate version.
+	version = "dev"
+	// Commit is the git commit hash that the build was generated from.
+	commit = "none"
+	// Date is the date the binary was produced.
+	date = "unknown"
 )
 
 // main is the entry point for the injector; creates an HTTPS listener and listing for v1.AdmissionReview requests
 func main() {
-	var certFile, keyFile, credentialsFile string
+	if err := Run(os.Args); err != nil {
+		fmt.Fprintf(os.Stderr, "%s\n", err)
+		os.Exit(exitFailure)
+	}
+	os.Exit(exitSuccess) // shouldn't hit this if run is invoked correctly
+}
 
-	flag.StringVar(&certFile, "cert", "tls/cert.pem", "the path of the public certificate file in PEM format")
-	flag.StringVar(&keyFile, "key", "tls/key.pem", "the path of the private key file in PEM format")
-	flag.StringVar(
-		&credentialsFile,
-		"credentials",
-		"credentials/config.json",
-		"the path of JSON formatted credentials file",
-	)
+// Run contains the actual invocation code for the injector and is public to allow running integration tests with it.
+func Run(args []string) error { //nolint:funlen,cyclop // ok for Run
+	log := logger.New()
+	log.Info().
+		Str("version", version).
+		Str("commit", commit).
+		Str("date", date).
+		Msg("syncer version information")
+
+	// Config is the configuration for the injector.
+	// This is provided by environment variables.
+	type Config struct {
+		CertFile            string `env:"DSV_CERT"  envDefault:"${HOME}/tls/cert.pem" envExpand:"true"`                       // Cert is the path to the public certificate file in PEM format.
+		KeyFile             string `env:"DSV_KEY" envDefault:"${HOME}/tls/key.pem" envExpand:"true"`                          // Key is the path to the private key file in PEM format.
+		CredentialsJSONFile string `env:"DSV_CREDENTIALS_JSON" envDefault:"${HOME}/credentials/config.json" envExpand:"true"` // CredentialsJSONFile is the path to the JSON formatted credentials file that is mounted as a secret.
+		ServerAddress       string `env:"DSV_SERVER_ADDRESS" envDefault:":18543"`                                             // ServerAddress is the address to listen on, e.g., 'localhost:8080' or ':8443'
+		Debug               bool   `env:"DSV_DEBUG" envDefault:"false"`                                                       // Debug enables debug logging.
+	}
 
-	server := new(http.Server)
+	cfg := Config{}
+	err := env.Parse(&cfg)
+	if err != nil {
+		log.Error().Err(err).Msg("unable to parse environment variables")
+	}
+	if cfg.Debug {
+		logger.EnableDebug()
+		log.Info().Msg("debug logging enabled")
+	}
+	log.Info().Strs("args", args).Msg("starting syncer, args passed, but not used, as environment variables are used instead")
```

Comment:
"args passed, but not used", but it may happen that no args were passed, right?

Reply:
oh no, those are just file names, soooorry

Reply:
I left default behavior to pass args but we aren't using. I can clean that up in a future PR if it makes sense, but will mark resolved as behavior known and want to get updates out asap.